staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go

```diff
@@ -238,10 +238,11 @@ func (c *Cloud) ensureLoadBalancerv2(namespacedName types.NamespacedName, loadBa
 						}
 					}
 
-					// recreate targetGroup if trafficPort or protocol changed
+					// recreate targetGroup if trafficPort or protocol changed, or if healthCheckProtocol changed from TCP
 					targetGroupRecreated := false
 					targetGroup, ok := nodePortTargetGroup[nodePort]
-					if !ok || aws.StringValue(targetGroup.Protocol) != mapping.TrafficProtocol {
+					invalidHealthCheckProtocolModification := aws.StringValue(targetGroup.HealthCheckProtocol) == elbv2.ProtocolEnumTcp && aws.StringValue(targetGroup.HealthCheckProtocol) != mapping.HealthCheckProtocol
```

Comment:
As per your explanation in the description:

```
This fix adds a check in ensureLoadBalancerv2 to recreate the target group if the HealthCheckProtocol is changing from TCP to anything else
```

- What happens to the existing target group ?
- How the traffic shaping happens ?
- If there are no traffic or anything why not just create the service with new load balancer with the right protocol and health check configuration ?